examples/echo_client_with_mam.py

```diff
@@ -0,0 +1,206 @@
+
+    """
+    A simple SleekXMPP bot that will echo messages it
+    receives, along with a short thank you message.
+    """
+
+    def __init__(self, jid, password):
+        sleekxmpp.ClientXMPP.__init__(self, jid, password)
+
+        # The session_start event will be triggered when
+        # the bot establishes its connection with the server
+        # and the XML streams are ready for use. We want to
+        # listen for this event so that we we can initialize
+        # our roster.
+        self.add_event_handler("session_start", self.start)
+
+        # The message event is triggered whenever a message
+        # stanza is received. Be aware that that includes
+        # MUC messages and error messages.
+        self.add_event_handler("message", self.message)
+
+    def start(self, event):
+        """
+        Process the session_start event.
+
+        Typical actions for the session_start event are
+        requesting the roster and broadcasting an initial
+        presence stanza.
+
+        Arguments:
+            event -- An empty dictionary. The session_start
+                     event does not provide any additional
+                     data.
+        """
+        self.send_presence()
+        self.get_roster()
+        
+        
+        print("Start MAM")
+        ret = self["xep_0313"].get_preferences()
+        print( "Archiving default is: ", ret["mam_prefs"]["default"] )
+        #self["xep_0313"].set_preferences(default="always", block=True)
+
+        ## async non-blocking
+        #answer = self["xep_0313"].retrieve(callback=self.__handle_mam_result)
+        
+        ## blocking
+        answer = self["xep_0313"].retrieve(callback=None, 
+                                           #start="2017-06-05T12:00:00+00:00",
+                                           #end="2017-06-05T23:59:59+00:00",
+                                           number_of_queried_elements=1,
+                                           collect_all=True)
+        
+        ## If no callback is used, the handler function is called here to display the results.
+        self.__handle_mam_result(answer)
+        
+        ## Note that a blocking call will finish before this,
+        #  an async call will finish after this.
+        print("End MAM")
+        
+        
+    def __handle_mam_result(self, response):
+        print( "Answer for query {}".format(response['mam_answer']['queryid']) )
+        print( "  - answer is: {}".format("complete" if response["mam_answer"]["complete"] else "incomplete") )
+        print( "  - last message is: {}".format(response["mam_answer"]["rsm"]["last"]) )
+        
+        result = response['mam_answer']['results']
+        
+        for x in result:
+            msg = x["mam_result"]["forwarded"]["message"]
+            delay = x["mam_result"]["forwarded"]["delay"]
+            if ( msg["body"] ):
+                print("==========")
+                print( "From:", msg["from"] )
```

Comment:
PEP-0008 recommends to not put spaces inside of parentheses, same for the next four lines.